File: app/src/main/java/com/bkjcb/rqapplication/treatment/fragment/TreatmentBackFragment.java
package com.bkjcb.rqapplication.treatment.fragment;

import android.text.TextUtils;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.bkjcb.rqapplication.base.MyApplication;
import com.bkjcb.rqapplication.R;
import com.bkjcb.rqapplication.base.fragment.BaseSimpleFragment;
import com.bkjcb.rqapplication.treatment.model.DefectDetail;
import com.bkjcb.rqapplication.treatment.model.DefectTreatmentModel;

import butterknife.BindView;

/**
 * Created by DengShuai on 2020/6/12.
 * Description :
 */
public class TreatmentBackFragment extends BaseSimpleFragment {


    @BindView(R.id.info_time)
    TextView mInfoTime;
    @BindView(R.id.info_accident_type)
    TextView mInfoAccidentType;
    @BindView(R.id.info_opinion)
    TextView mInfoOpinion;
    @BindView(R.id.info_name)
    TextView mInfoName;
    @BindView(R.id.info_address)
    TextView mInfoAddress;
    @BindView(R.id.record_remark)
    EditText mRecordRemark;
    protected DefectTreatmentModel model;
    protected DefectDetail result;
    protected boolean isCanChange = true;

    public DefectDetail getResult() {
        return result;
    }

    public void setResult(DefectDetail result) {
        this.result = result;
    }

    public void setModel(DefectTreatmentModel model) {
        this.model = model;
    }

    public static TreatmentBackFragment newInstance(DefectTreatmentModel model) {
        TreatmentBackFragment fragment = new TreatmentBackFragment();
        fragment.setModel(model);
        return fragment;
    }

    public static TreatmentBackFragment newInstance(DefectTreatmentModel model, DefectDetail result) {
        TreatmentBackFragment fragment = new TreatmentBackFragment();
        fragment.setModel(model);
        fragment.setResult(result);
        return fragment;
    }

    @Override
    public void setResId() {
        resId = R.layout.fragment_treatment_back;
    }

    @Override
    protected void initView() {
        if (model != null) {
            mInfoTime.setText(model.getUnitDisposalTime());
            mInfoAccidentType.setText(model.getCasesType());
            mInfoOpinion.setText(model.getOpinions());
            mInfoName.setText(model.getUserName());
            mInfoAddress.setText(model.getUserAddress());
        }
    }

    @Override
    protected void initData() {
        if (result == null) {
            result = new DefectDetail();
            result.setUserId(MyApplication.getUser().getUserId());
            result.setMtfId(model.getMtfId());
        } else {
            isCanChange = false;
            setText(mRecordRemark,result.getJzReasons());
        }
    }

    protected void collectParams() {
        result.setJzReasons(getText(mRecordRemark));
        result.setType("2");
        result.setRemotePath("");
    }

    protected boolean verifyData() {
        return verify(mRecordRemark, result.getJzReasons(), "请填写退单理由！");
    }

    protected boolean verify(View view, String value, String tip) {
        if (TextUtils.isEmpty(value)) {
            if (view != null) {
                view.requestFocus();
            }
            Toast.makeText(getContext(), tip, Toast.LENGTH_SHORT).show();
            return false;
        }
        return true;
    }

    protected String getText(TextView view) {
        return view.getText().toString();
    }

    protected void setText(TextView view, String value) {
        view.setText(value);
        view.setEnabled(isCanChange);
    }

    public DefectDetail prepareSubmit() {
        collectParams();
        return verifyData() ? result : null;
    }
}

File: app/src/main/java/com/bkjcb/rqapplication/treatmentdefect/fragment/TreatmentBackFragment.kt
package com.bkjcb.rqapplication.treatmentdefect.fragment

import android.text.TextUtils
import android.view.View
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import butterknife.BindView
import com.bkjcb.rqapplication.MyApplication
import com.bkjcb.rqapplication.R
import com.bkjcb.rqapplication.fragment.BaseSimpleFragment
import com.bkjcb.rqapplication.treatmentdefect.model.DefectDetail
import com.bkjcb.rqapplication.treatmentdefect.model.DefectTreatmentModel
import kotlinx.android.synthetic.main.fragment_treatment_back.*

/**
 * Created by DengShuai on 2020/6/12.
 * Description :
 */
open class TreatmentBackFragment : BaseSimpleFragment() {

    protected var model: DefectTreatmentModel? = null
    var result: DefectDetail? = null
    protected var isCanChange = true

    override fun initView() {
        if (model != null) {
            info_time.text = model!!.processTime
            info_accident_type.text = model!!.casesType
            info_opinion.text = model!!.opinions
            info_name.text = model!!.userName
            info_address.text = model!!.userAddress
        }
    }

    override fun initData() {
        if (result == null) {
            result = DefectDetail()
            result!!.userId = MyApplication.getUser().userId
            result!!.mtfId = model!!.mtfId
        }
    }

    protected open fun collectParams() {
        result!!.jzReasons = getText(record_remark)
        result!!.type = "2"
        result!!.remotePath = ""
    }

    protected open fun verifyData(): Boolean {
        return verify(record_remark, result!!.jzReasons, "请填写退单理由！")
    }

    protected fun verify(view: View?, value: String?, tip: String?): Boolean {
        if (TextUtils.isEmpty(value)) {
            view?.requestFocus()
            Toast.makeText(context, tip, Toast.LENGTH_SHORT).show()
            return false
        }
        return true
    }

    protected fun getText(view: TextView?): String {
        return view?.text.toString()
    }

    protected fun setText(view: TextView, value: String?) {
        view.text = value
        view.isEnabled = isCanChange
    }

    fun prepareSubmit(): DefectDetail? {
        collectParams()
        return if (verifyData()) result else null
    }

    override fun initResID(): Int {
        return R.layout.fragment_treatment_back
    }

    companion object {
        fun newInstance(model: DefectTreatmentModel?): TreatmentBackFragment {
            val fragment = TreatmentBackFragment()
            fragment.model = model
            return fragment
        }
    }
}